owsi-core/owsi-core-components/owsi-core-component-commons/src/main/java/fr/openwide/core/commons/util/binding/BindingUtils.java

```diff
@@ -0,0 +1,20 @@
+package fr.openwide.core.commons.util.binding;
+
+import org.bindgen.Binding;
+
+public final class BindingUtils {
+
+	public static Class<?> getRootType(Binding<?> binding) {
```

Comment:
In fact, I think we're using AbstractBinding in our APIs right now, at least in the diff APIs. So there probably won't be any problem with this change: every caller has an AbstractBinding to give us, which very probably means it has an AbstractCoreBinding, or that it could ask its own caller an AbstractCoreBinding. I think we don't use interfaces very much at the moment.

Now, to be cleaner, we should probably have used BindingRoot instead of AbstractBinding in our APIs in the first place. And, if we want this change to be cleaner, we could introduce a CoreBindingRoot interface that extends BindingRoot and adds the methods we will implement in AbstractCoreBinding.

I just checked, and the only bindings that do not extend AbstractCoreBinding are those extending GenericObjectBindingPath, which bindgen should allow to customize just like the root class for bindings (it probably wouldn't be very hard to patch), and which is rarely used anyway.